percona/controller/pgbackup/controller.go

```diff
@@ -228,11 +230,22 @@ func (r *PGBackupReconciler) Reconcile(ctx context.Context, request reconcile.Re
 			log.Info("Waiting for backup to complete")
 			return reconcile.Result{RequeueAfter: time.Second * 5}, nil
 		}
-		finished, err := runFinalizers(ctx, r.Client, pgBackup)
+
+		pgCluster := &v2.PerconaPGCluster{}
+		if err := r.Client.Get(ctx, types.NamespacedName{Name: pgBackup.Spec.PGCluster, Namespace: request.Namespace}, pgCluster); err != nil {
+			if !k8serrors.IsNotFound(err) {
+				return reconcile.Result{}, errors.Wrap(err, "get PostgresCluster")
+			}
+			pgCluster = nil
```

Comment:
Commenting only to confirm that I understand this fully. What we want to do with assigning this to nil is effectively to be able to remove the finalizer when `RunFinalizer` is executed

```
	if err := f(ctx, obj); err != nil {
		if errors.Is(err, ErrFinalizerPending) {
			return false, nil
		}
		return false, errors.Wrapf(err, "run finalizer %s", finalizer)
	}

	if controllerutil.RemoveFinalizer(obj, finalizer) {
		log.Info("Removing finalizer", "name", finalizer)
		if err := cl.Patch(ctx, obj, client.MergeFrom(orig)); err != nil {
			return false, errors.Wrap(err, "remove finalizers")
		}
	}
```


and more specifically, when ☝🏽 these are executed, is that correct?